### ISSUE TITLE
oAuth token generation endpoint should support JSON format

### ISSUE THREAD

**Issue by buger** [TykTechnologies/tyk#2375]
**Do you want to request a *feature* or report a *bug*?**
feature
**What is the current behavior?**
At the moment we do support only Form values, e.g. `grant_type=client_credentials&client_id=...&client_secret=...`
**What is the expected behavior?**
Support JSON format if Content-Type set to application/json:
`{"client_id":"...","client_secret":"...", "grant_type":"client_credentials"}`
**Which versions of Tyk affected by this issue? Did this work in previous versions of Tyk?**
2.8

**Comment by letzya** [comment]
@buger https://github.com/openshift/osin/issues/163 FYI

**Comment by ilijabojanovic** [comment]
@gernest 

On latest master(1701e40fd4172ebb54fa3056d9f3859c80b819fe) :

When we sent json to /oauth/token endpoint server panics

Example:
```
curl -X POST \
  https://www.tyk-test.com:8181/api-oauth/oauth/token \
  -H 'Content-Type: application/json' \
  -H 'Postman-Token: f85f30fb-52eb-4ebd-8d15-0b45de8d79b3' \
  -H 'cache-control: no-cache' \
  -d '{
	"grant_type": "client_credentials",
	"client_id": "39165fdd9d374d99934ae6902642c95a",
	"client_secret": "OWNlZDQxNjMtNDcwMC00OTVlLWFjODMtMmRhMDgwYzBiYWEx"
}'
```

Here is console log:
```
2019/07/24 11:16:23 http: panic serving 127.0.0.1:50975: runtime error: invalid memory address or nil pointer dereference
goroutine 67687 [running]:
net/http.(*conn).serve.func1(0xc0002b1540)
	/usr/local/go/src/net/http/server.go:1746 +0xd0
panic(0x1b98720, 0x2993950)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/TykTechnologies/tyk/gateway.JSONToFormValues(0xc00b844400, 0x0, 0x0)
	/Users/ilijabojanovic/work/src/github.com/TykTechnologies/tyk/gateway/oauth_manager.go:262 +0xc6
github.com/TykTechnologies/tyk/gateway.(*OAuthManager).HandleAccess(0xc00c2e2a50, 0xc00b844400, 0xc)
	/Users/ilijabojanovic/work/src/github.com/TykTechnologies/tyk/gateway/oauth_manager.go:290 +0x78
github.com/TykTechnologies/tyk/gateway.(*OAuthHandlers).HandleAccessRequest(0xc00c2e2a50, 0x2031240, 0xc0074b9c00, 0xc00b844400)
	/Users/ilijabojanovic/work/src/github.com/TykTechnologies/tyk/gateway/oauth_manager.go:184 +0xa4
github.com/TykTechnologies/tyk/gateway.(*OAuthHandlers).HandleAccessRequest-fm(0x2031240, 0xc0074b9c00, 0xc00b844400)
	/Users/ilijabojanovic/work/src/github.com/TykTechnologies/tyk/gateway/server.go:467 +0x48
github.com/TykTechnologies/tyk/gateway.allowMethods.func1(0x2031240, 0xc0074b9c00, 0xc00b844400)
	/Users/ilijabojanovic/work/src/github.com/TykTechnologies/tyk/gateway/api.go:135 +0x166
github.com/TykTechnologies/tyk/gateway.addSecureAndCacheHeaders.func1(0x2031240, 0xc0074b9c00, 0xc00b844400)
	/Users/ilijabojanovic/work/src/github.com/TykTechnologies/tyk/gateway/api.go:127 +0x265
net/http.HandlerFunc.ServeHTTP(0xc00c2e2c20, 0x2031240, 0xc0074b9c00, 0xc00b844400)
	/usr/local/go/src/net/http/server.go:1964 +0x44
github.com/TykTechnologies/tyk/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc00dcef5c0, 0x2031240, 0xc0074b9c00, 0xc00b844400)
	/Users/ilijabojanovic/work/src/github.com/TykTechnologies/tyk/vendor/github.com/gorilla/mux/mux.go:150 +0xf1
github.com/TykTechnologies/tyk/gateway.mainHandler.ServeHTTP(0x2031240, 0xc0074b9c00, 0xc00b844200)
	/Users/ilijabojanovic/work/src/github.com/TykTechnologies/tyk/gateway/server.go:1253 +0xa5
net/http.serverHandler.ServeHTTP(0xc00066c1a0, 0x2031240, 0xc0074b9c00, 0xc00b844200)
	/usr/local/go/src/net/http/server.go:2741 +0xab
net/http.(*conn).serve(0xc0002b1540, 0x2032740, 0xc00e6285c0)
	/usr/local/go/src/net/http/server.go:1847 +0x646
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2851 +0x2f5
```

**Comment by gernest** [comment]
@ilijabojanovic I was able to reproduce this, it was my mistake I had tests which didn't cover one scenario.

The fix is #2434  .
